R/tm_missing_data.R

```diff
@@ -348,11 +348,44 @@ srv_missing_data <- function(id, data, reporter, filter_panel_api, dataname, par
   checkmate::assert_class(data, "tdata")
   moduleServer(id, function(input, output, session) {
     prev_group_by_var <- reactiveVal("")
-
     data_r <- data[[dataname]]
-
     data_keys <- reactive(get_join_keys(data)$get(dataname)[[dataname]])
 
+    iv_r <- reactive({
+      iv <- shinyvalidate::InputValidator$new()
+      iv$add_rule(
+        "variables_select",
+        shinyvalidate::sv_required("At least one reference variable needs to be selected.")
+      )
+      iv$add_rule(
+        "variables_select",
+        ~ if (length(setdiff((.), data_keys())) < 1) "Please also select non-key columns."
+      )
+      iv_summary_table <- shinyvalidate::InputValidator$new()
+      iv_summary_table$condition(~ input$summary_type == "By Variable Levels")
+      iv_summary_table$add_rule("count_type", shinyvalidate::sv_required("Please select type of counts"))
+      iv_summary_table$add_rule(
+        "group_by_vals",
+        shinyvalidate::sv_required("Please select both group-by variable and values")
+      )
+      iv_summary_table$add_rule(
+        "group_by_var",
+        ~ if (length(.) > 0 && length(input$variables_select) == 1 && (.) == input$variables_select) {
+          "If only one reference variable is selected it must not be the grouping variable."
+        }
+      )
+      iv_summary_table$add_rule(
+        "variables_select",
```

Comment:
One variable must not be validated by multiple validators as conflicts may ensue. (I'm not sure if this includes children.) I suggest using `crule` here.

Reply:
A condition is set for `iv_summary_table`:
`iv_summary_table$condition(~ input$summary_type == "By Variable Levels")`

Reply:
What I mean is there is a rule for `variables_select` in `iv` AND in `iv_summary_table`. Even though this is a child, it is another validator. `shinyvalidate` strongly discourages putting rules for one input in several validators. If it works here, I don't mind much.